### PR TITLE
fix: use generated tsconfig and virtual file augmentations

### PR DESCRIPTION
**Pull request by BobbieGoede**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
This cleans up the virtual file augmentations as well as the tsconfig file to resemble the module started more.

I had to add a bunch of type assertions for both `I18n` and runtime config types, will need to look into these,

Thanks to @userquin for giving examples on how to deal with the virtual file types!

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -180,8 +185,9 @@ export default defineNuxtModule<NuxtI18nOptions>({
     addPlugin(resolve(runtimeDir, 'plugins/i18n'))
     addPlugin(resolve(runtimeDir, 'plugins/switch-locale-path-ssr'))
 
+    const resolver = createResolver(import.meta.url)
     // for composables
-    nuxt.options.alias['#i18n'] = resolve(distDir, 'runtime/composables/index.js')
+    nuxt.options.alias['#i18n'] = resolver.resolve('./runtime/composables/index')
```
Comment (userquin) [inline src/module.ts:190]: add `.js` extension here

File: package.json [hunk 2]
```diff
@@ -47,17 +50,18 @@
...
   "files": [
-    "dist"
+    "dist",
+    "internals.d.ts"
   ],
   "scripts": {
     "prepare": "git config --local core.hooksPath .githooks",
-    "build": "nuxi build-module .",
-    "stub": "nuxi build-module --stub .",
+    "build": "nuxt-module-build build .",
```
Comment (userquin) [inline package.json]: This script should be `nuxt-module-build prepare && nuxt-module-build build .`, you need to create `.nuxt` stuff before running other scripts

**Review by kazupon — COMMENTED** [review]
Great work!
This approach makes sense!

I have always found it strange that the `src` directory contains a mixture of `.ts` implementation files and `.d.ts` type definition files, and I have wanted to refactor them at some point.

Virtual modules solve this issue, but it depends on the bundler. Since the nuxt module is a framework module system and Nuxt provides a bundling framework system, I believe the bundler dependency is not an issue.

Let's keep it up!

@userquin 
Thank you for supporting!

File: src/module.ts (hunk 1 above)
Comment (BobbieGoede) [reply src/module.ts:190]: Looks like adding the extension breaks explicit imports (when having auto imports disabled) https://github.com/nuxt-modules/i18n/actions/runs/10947564325/job/30397002095?pr=3112

**Comment by BobbieGoede** [comment]
Seems to be working as expected now 🤔 It's unfortunate that we now have two identical virtual file augmentations, and will have to look into the added type assertions in a future refactor. But overall this should make the module types more predictable and in line with established (but undocumented 😅) conventions among Nuxt modules.

**Pull request merged**